joy_teleop/test/test_action_fibonacci.py

```diff
@@ -0,0 +1,136 @@
+    parameters = {}
+    parameters['fibonacci.type'] = 'action'
+    parameters['fibonacci.interface_type'] = 'action_tutorials_interfaces/action/Fibonacci'
+    parameters['fibonacci.action_name'] = '/fibonacci'
+    parameters['fibonacci.buttons'] = [2]
+    parameters['fibonacci.action_goal'] = {'order': 5}
+    joy_teleop_node = launch_ros.actions.Node(
+        package='joy_teleop',
+        node_executable='joy_teleop',
+        output='both',
+        parameters=[parameters])
+
+    return (
+        launch.LaunchDescription([
+            joy_teleop_node,
+            launch_testing.actions.ReadyToTest(),
+        ]),
+        {
+            'joy_teleop_node': joy_teleop_node,
+        }
+    )
+
+
+class TestJoyTeleopActionFibonacci(unittest.TestCase):
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.node = rclpy.create_node('test_action_fibonacci', context=self.context)
+        self.executor = rclpy.executors.SingleThreadedExecutor(context=self.context)
+        self.executor.add_node(self.node)
+
+    def tearDown(self):
+        self.node.destroy_node()
+        rclpy.shutdown(context=self.context)
+
+    def test_simple_message(self, launch_service, joy_teleop_node, proc_output):
+        joy_publisher = self.node.create_publisher(
+            msg_type=sensor_msgs.msg.Joy,
+            topic='joy',
+            qos_profile=1,
+        )
+
+        sequence = []
+        future = rclpy.task.Future()
+
+        def fibonacci_callback(goal_handle):
+            nonlocal future
+
+            sequence.append(0)
+            sequence.append(1)
+            for i in range(1, goal_handle.request.order):
+                sequence.append(sequence[i] + sequence[i-1])
+
+            goal_handle.succeed()
+            result = action_tutorials_interfaces.action.Fibonacci.Result()
+            result.sequence = sequence
+            future.set_result(True)
+            return result
+
+        action_server = rclpy.action.ActionServer(
+            self.node,
+            action_tutorials_interfaces.action.Fibonacci,
+            'fibonacci',
+            fibonacci_callback)
+
+        joy_msg = sensor_msgs.msg.Joy()
+        joy_msg.header.stamp.sec = 0
+        joy_msg.header.stamp.nanosec = 0
+        joy_msg.header.frame_id = ''
+        joy_msg.axes = []
+        # Above we set the button to be used as '2', so here we set the '2' button active.
+        joy_msg.buttons = [0, 0, 1]
+
+        def publish_message():
+            joy_publisher.publish(joy_msg)
+            joy_msg.buttons[2] = int(not joy_msg.buttons[2])
+        publish_timer = self.node.create_timer(0.1, publish_message)
+
+        self.executor.spin_until_future_complete(future, timeout_sec=10)
+
+        # Check
+        assert future.done() and future.result(), 'Timed out waiting for action to complete'
+        self.assertTrue(sequence == [0, 1, 1, 2, 3, 5])
```

Comment:
@clalancette why mixing both styles?

Reply:
Oh, good question.  I wrote the code at different times, and I just didn't realize.  Which one is preferred?

Reply:
In `unittest`-style tests, IIRC `self.assertTrue` is preferred.

Reply:
Ah yes, thanks.  I just read the docs, and the difference is that unittest collects failures for `assertTrue` and reports all of them, while `assert` just terminates immediately.  So yeah, I think `assertTrue` is better; will switch to that.